src/core/algorithms/gfd/gfd_miner.cpp

```diff
@@ -0,0 +1,671 @@
+#include "gfd_miner.h"
+
+#include <numeric>
+
+#include <boost/graph/copy.hpp>
+#include <boost/graph/isomorphism.hpp>
+#include <boost/graph/vf2_sub_graph_iso.hpp>
+#include <easylogging++.h>
+
+#include "config/option_using.h"
+
+namespace {
+
+using namespace algos;
+
+using Embedding = std::map<vertex_t, vertex_t>;
+using Embeddings = std::vector<Embedding>;
+
+using SingleRule = std::pair<std::vector<Literal>, Literal>;
+using Rule = std::pair<std::vector<Literal>, std::vector<Literal>>;
+using Rules = std::vector<Rule>;
+
+using Info = std::map<std::string, std::set<std::string>>;
+
+void NextSubset(std::vector<std::size_t>& indices, std::size_t const border) {
+    if (indices.at(0) == border - indices.size()) {
+        indices = {};
+        return;
+    }
+    std::size_t index = 0;
+    for (std::size_t i = indices.size() - 1; i >= 0; --i) {
```

Comment:
As I said before, it is a potential infinite cycle
https://github.com/Desbordante/desbordante-core/pull/465#discussion_r1790848369